packages/react-vapor/src/components/select/hoc/SelectWithPredicate.tsx

```diff
@@ -46,7 +46,7 @@ export const selectWithPredicate = <P extends Omit<ISelectOwnProps, 'button'> &
         const {onUpdate, predicate} = props;
         React.useEffect(() => {
             onUpdate?.();
-        }, [predicate, onUpdate]);
+        }, [predicate]);

```

Comment:
are you removing all values from the initiator array that may be undefined?

Reply:
every function from mapDispatchToProps